execution/gethexec/timeboost_sequencer.go

```
@@ -0,0 +1,549 @@
+			// For now we are logging
+			for _, queueItem := range queueItems {
+				log.Error("Error processing transaction", "err", sequencerInternalError, "queueItem", queueItem)
+			}
+		}
+
+		returnValue = true
+	}()
+
+	lastBlock := s.execEngine.bc.CurrentBlock()
+	config := s.config()
+
+	for {
+		var queueItem timeboostTransactionQueueItem
+		done := false
+		//  Transaction retry queue should only
+		//  have transactions from a given round id
+		if s.txRetryQueue.Len() > 0 {
+			queueItem = s.txRetryQueue.Pop()
+		} else if s.txQueue.Len() == 0 {
+			// This means we have no transactions in the txRetryQueue and
+			// we also dont have any sailfish rounds to process
+			done = true
+		} else {
+			// Only add transactions from the same round id or if the queue is empty
+			if queueItems == nil {
+				queueItems = make([]timeboostTransactionQueueItem, 0)
+				queueItem = s.txQueue.Pop()
+			} else if queueItems[len(queueItems)-1].roundId == s.txQueue.Peek().roundId {
+				queueItem = s.txQueue.Pop()
+			} else {
+				done = true
+			}
+		}
+
+		// If context is done, return false
+		select {
+		case <-ctx.Done():
+			return false
+		default:
+		}
+
+		if done {
+			// We have added as many transactions as we can
+			break
+		}
+
+		if queueItem.txSize > s.config().MaxTxDataSize {
+			// This tx is too large
+			// Even if its a priority item this should be skipped,
+			// TODO: return the error to the user here
+			log.Warn("timeboost transaction is too large", "txSize", queueItem.txSize, "maxTxDataSize", s.config().MaxTxDataSize, "hash", queueItem.tx.Hash().Hex())
+			continue
+		}
+
+		if arbmath.BigLessThan(queueItem.tx.GasFeeCap(), lastBlock.BaseFee) {
+			// This tx is too low gas fee
+			// TODO: return the error to the user here
+			log.Warn("timeboost transaction has too low gas fee", "txSize", queueItem.txSize, "gasFeeCap", queueItem.tx.GasFeeCap(), "baseFee", lastBlock.BaseFee, "hash", queueItem.tx.Hash().Hex())
+			continue
+		}
```

Can we move this part to `preTxFilter`? This loop is too long and distracts readers

---

The reason why its placed here is because if a transaction is greater than max size or gas we need to return an error to the user **as soon as possible** but currently we still dont have a way to return back an error to the user.  This design is similar to how nitro sequencer works and these checks in nitro sequencer also happen before the transactions are added to the queue

`preTxFilter` happens on a queue which is ready to create a block and we are doing some final checks.